test/gtest/ucp/test_ucp_tag_xfer.cc

```diff
@@ -1152,55 +1161,55 @@ UCP_INSTANTIATE_TEST_CASE(test_ucp_tag_stats)
 
 class multi_rail_max : public test_ucp_tag_xfer {
 public:
-    void init()
+    void init() override
     {
         stats_activate();
         test_ucp_tag_xfer::init();
     }
 
-    void cleanup()
+    void cleanup() override
     {
         test_ucp_tag_xfer::cleanup();
         stats_restore();
     }
-};
 
-UCS_TEST_P(multi_rail_max, max_lanes, "IB_NUM_PATHS?=8", "TM_SW_RNDV=y")
-{
-    /* MAX_RNDV_LANES must be initialized after 'init' */
-    modify_config("MAX_RNDV_LANES", "8");
-    entity *e1 = create_entity(true);
-    entity *e2 = create_entity(false);
+    uint64_t get_bytes_sent(ucp_ep_h ep, int lane_index)
+    {
+        uct_base_ep_t *lane = ucs_derived_of(ucp_ep_get_lane(ep, lane_index),
+                                             uct_base_ep_t);
 
-    e1->connect(e2, get_ep_params());
-    e2->connect(e1, get_ep_params());
-    test_run_xfer(true, true, true, true, false);
+        return UCS_STATS_GET_COUNTER(lane->stats, UCT_EP_STAT_BYTES_SHORT) +
+               UCS_STATS_GET_COUNTER(lane->stats, UCT_EP_STAT_BYTES_BCOPY) +
+               UCS_STATS_GET_COUNTER(lane->stats, UCT_EP_STAT_BYTES_ZCOPY);
+    }
 
-    if(RUNNING_ON_VALGRIND) {
-        UCS_TEST_SKIP_R("RUNNING_ON_VALGRIND");
+    uint32_t num_lanes() override
+    {
+        /* Init max lanes for the test */
+        return max_lanes;
     }
 
-    ucp_lane_index_t num_lanes = ucp_ep_num_lanes(e2->ep());
-    ASSERT_EQ(ucp_ep_num_lanes(e1->ep()), num_lanes);
-    ASSERT_EQ(num_lanes, 8);
+    const uint32_t max_lanes = 8;
+};
 
-    for (int i = 0; i < num_lanes; ++i) {
-        uct_base_ep_t *lane = ucs_derived_of(ucp_ep_get_lane(e1->ep(), i),
-                                             uct_base_ep_t);
-        uint64_t bytes_sent =
-                UCS_STATS_GET_COUNTER(lane->stats, UCT_EP_STAT_BYTES_SHORT) +
-                UCS_STATS_GET_COUNTER(lane->stats, UCT_EP_STAT_BYTES_BCOPY) +
-                UCS_STATS_GET_COUNTER(lane->stats, UCT_EP_STAT_BYTES_ZCOPY);
+UCS_TEST_P(multi_rail_max, max_lanes, "IB_NUM_PATHS?=8", "TM_SW_RNDV=y",
+           "RNDV_THRESH=1", "MIN_RNDV_CHUNK_SIZE=1")
+{
+    receiver().connect(&sender(), get_ep_params());
+    test_run_xfer(true, true, true, true, false);
 
-        lane = ucs_derived_of(ucp_ep_get_lane(e2->ep(), i), uct_base_ep_t);
+    uint32_t msg_size_threshold = 50000 / ucs::test_time_multiplier();
```

Comment:
remove this var, or move it right before ASSERT_GE
rename to something like min_expected_bytes